eng/testing/tests.targets

```diff
@@ -136,6 +132,23 @@
       <RunScriptOutputPath>$([MSBuild]::NormalizePath('$(RunScriptOutputDirectory)', '$(RunScriptOutputName)'))</RunScriptOutputPath>
     </PropertyGroup>
 
+    <PropertyGroup Condition="'$(IsXUnitLogCheckerSupported)' == 'true' and
+                              '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                              $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
+      <SetScriptCommand Condition="'$(TargetOS)' == 'windows'">$(SetScriptCommand)
+set __TestArchitecture=$(TargetArchitecture);
+set __IsXUnitLogCheckerSupported=1
+      </SetScriptCommand>
+      <SetScriptCommand Condition="'$(TargetOS)' != 'windows'">$(SetScriptCommand)
+export __TestArchitecture=$(TargetArchitecture);
+export __IsXUnitLogCheckerSupported=1
+      </SetScriptCommand>        
```

Comment:
These should be items instead of properties.

Reply:
Ok sure, I can use items. Note that there's lots of examples, especially with helix-related data, where we use a property first to collect all the elements, and then store them into a single item.

Reply:
This isn't necessary here. I.e. here you can just inline the value instead of defining a separate property:

```
<PropertyGroup>
  <CompileItem>A.cs</CompileItem>
</PropertyGroup>

<ItemGroup>
  <Compile Include="$(CompileItem)" />
</ItemGroup>
```

Reply:
Pushed the commit.

Reply:
I confirmed the generated runner script contains the env variables as expected.